src/staticdata.c

```diff
@@ -2384,6 +2387,7 @@ static void strip_specializations_(jl_method_instance_t *mi)
                 if (jl_atomic_cmpswap_relaxed(&codeinst->inferred, &inferred, stripped)) {
                     jl_gc_wb(codeinst, stripped);
                 }
+                record_field_change((jl_value_t**)&codeinst->debuginfo, (jl_value_t*)jl_nulldebuginfo);
             }
         }
```

Comment:
```suggestion
            }
        }
        if (jl_options.strip_metadata)
            record_field_change((jl_value_t**)&codeinst->debuginfo, (jl_value_t*)jl_nulldebuginfo);
```

Even if the `inferred` field is nothing, we should still clear out the `debuginfo`